### PR TITLE
React 16.6 compatibility 

### PR DESCRIPTION
**Pull request by tadeuszwojcik**
Improvements and fixes of React v16.6 context handling when rendering server side.

**Other information:**

Closes https://github.com/deity-io/falcon/issues/112 , https://github.com/deity-io/falcon/issues/213

## Detailed changelog

- (falcon-client, shop-with-blog/client): replaced `react-async-component` and `react-async-bootstrapper2` packages with `@loadable` components (https://www.smooth-code.com/open-source/loadable-components/) - fixes issue with context handling in React 16.6 - not more react tree traversal on the server  🚀 
- (falcon-client, shop-with-blog/client, ecommerce-uikit): updated `react-apollo` to v2.3.2 that doesn't have custom react tree traversal logic on the server - uses `renderToString` - fixes issue with context handling in React 16.6 🚀 
- link prefetch/preload support added (https://medium.com/webpack/link-rel-prefetch-preload-in-webpack-51a52358f84c ) 🚀 

- updated `react` and `react-dom` dependencies to **v16.6.3** in all packages 🚀 💥 

- updated `react-router-dom` to `v4.4.0-beta.6` - previous version was producing warning with loadable components (https://github.com/ReactTraining/react-router/issues/6497)

- updated webpack to `v4.24.0` - previous versions weren't compatible with `@loadable` components

- fixed issue in `falcon-ui` with SSR rendering of `Details` component 🐛 

- added error boundary in `shop-with-blog\client` - `@loadable` components don't have error handler (as we had in react-async-component) 🆕 

- faster SSR - instead of 5 `renderToString` calls we have only one now (plus whatever `react-apollo` does inside) 🚀 

**Before**

current-timings
**After**
new-timings

#### All those changes make falcon compatible with 16.7.alpha as well ( **hooks** 🎉 )  

Remaining issues:
- I've found issue with `@loadable` components that was causing it's webpack plugin to hang in production mode -  https://github.com/smooth-code/loadable-components/issues/179   and currently using workaround for it (that saves loadable-stats.json to `build/public` dir not `build`) - that workaround it seems to affect hot reloading logic a little bit - to many rebuilds from time to time, also sometimes after building for production and then again for dev it seems to load incorrect version, requires restart.
IMHO it's acceptable as usual workflow works fine and hopefully fix for that will be implemented soon in `@loadable` components (or I'll create pull request for that)

### CODE REVIEW

File: examples/shop-with-blog/client/src/theme.js [hunk 1]
```diff
@@ -206,3 +206,12 @@ export const deityGreenTheme = createTheme({
     }
   }
 });
+
+export const normalizeCss = {
```
Comment (mateuszwozniak) [inline examples/shop-with-blog/client/src/theme.js]: Let's rename it to `globalCss` so it's more clear what it's intended for.

Comment (tadeuszwojcik) [reply examples/shop-with-blog/client/src/theme.js]: Done.

File: UPGRADE.md [hunk 2]
```diff
@@ -6,11 +6,39 @@ on how to migrate your previously generated Falcon-based project to a newer vers
...
 
+## Falcon 0.2.0 to 0.3.0
+
+### Falcon-Client 0.3.0
+
+If you were using `Async` components, please change them `loadable` instead:
+
+```diff
+- import AsyncComponent from 'src/components/Async';
++ import loadable from 'src/components/loadable';
```
Comment (Borales) [inline UPGRADE.md]: @tadeuszwojcik could you please put a link to the file (from `master` probably) where developers could get this module from?

Comment (tadeuszwojcik) [reply UPGRADE.md]: done

**Pull request merged**